### PR TITLE
implementation of more member functions

### PR DESCRIPTION
**Pull request by kangmingtay**
add command to remove member from the member list

### CODE REVIEW

**Review by kianhong95 — CHANGES_REQUESTED** [review]
The three test methods: updateMemberPoints(), invalidMemberInOrder() and validMemberInOrder(), should assert some sort of expected output, otherwise the test would be redundant.

For updateMemberPoints() and validMemberInOrder() assert an expected output for them.

For invalidMemberInOrder(), if you don't want to use assertion can use something like:
@Test(expected = IndexOutOfBoundsException.class)
instead of throwing an Exception

**Review by kianhong95 — APPROVED** [review]

File: src/seedu/addressbook/data/Rms.java [hunk 1]
```diff
@@ -141,6 +141,23 @@ public boolean containsMember(ReadOnlyMember key) {
         return allMembers.contains(key);
     }
 
+    /**
+     *  Checks if a member in the order is in the list of members
+     *  Returns the member if found, else create a new Member using the data from the member in the order
+     */
+    public Member findMemberInOrder(Member orderMember) {
```
Comment (px1099) [inline src/seedu/addressbook/data/Rms.java]: ~~Can you change the parameter type to ReadOnlyMember ?~~

Comment (px1099) [reply src/seedu/addressbook/data/Rms.java]: ~~Better yet, you can create another overloading version that can except ReadOnlyMember instead~~

File: src/seedu/addressbook/data/Rms.java [hunk 2]
```diff
@@ -141,6 +141,23 @@ public boolean containsMember(ReadOnlyMember key) {
...
+        MemberName name = orderMember.getName();
+        Points points = orderMember.getPoints();
+        Date date = orderMember.getDate();
+
+        for(Member member : allMembers) {
+            if((name == member.getName()) && (points == member.getPoints()) && (date == member.getDate())) {
```
Comment (px1099) [inline src/seedu/addressbook/data/Rms.java]: you can make use of the isSameStateAs() method from ReadOnlyMember.java

File: src/seedu/addressbook/data/Rms.java (hunk 1 above)
Comment (px1099) [reply src/seedu/addressbook/data/Rms.java]: Changing the parameter type to ReadOnlyMember can cover both Member and ReadOnlyMember inputs

File: src/seedu/addressbook/data/member/MemberName.java [hunk 3]
```diff
@@ -10,57 +11,53 @@
...
 
 
-    public static final String EXAMPLE = "Eve Vee";
-    public static final String MESSAGE_NAME_CONSTRAINTS = "Member names should be spaces or alphanumeric characters";
-    public static final String NAME_VALIDATION_REGEX = "[\\p{Alnum} ]+";
-
-
-    public final String value;
+//    public static final String EXAMPLE = "Eve Vee";
+//    public static final String MESSAGE_NAME_CONSTRAINTS = "Member names should be spaces or alphanumeric characters";
```
Comment (px1099) [inline src/seedu/addressbook/data/member/MemberName.java]: (Unnecessary nitpicking) Should use /*   */ instead of using // for every line

Comment (kangmingtay) [reply src/seedu/addressbook/data/member/MemberName.java]: It is an intelliJ shortcut to block comment. Just highlight the area you wanna comment out and click [ ctrl + / ].

**Pull request merged**